pkg/build/android.go

```diff
@@ -65,33 +62,147 @@ func (a android) readCompiler(archivePath string) (string, error) {
+	if osutil.IsExist(gccPluginsDir) {
+		if err := a.writeFile(filepath.Join(gccPluginsDir, "randomize_layout_seed.h"), []byte(seed)); err != nil {
+			return err
+		}
+	}
+
+	// Different key is generated for each build if key is not provided.
+	// see Documentation/reproducible-builds.rst. This is causing problems to our signature calculation.
+	certsDir := filepath.Join(commonKernelDir, "certs")
+	if osutil.IsExist(certsDir) {
+		if err := a.writeFile(filepath.Join(certsDir, "signing_key.pem"), []byte(moduleSigningKey)); err != nil {
+			return err
+		}
+	}
+
+	// Make kernel image and prepare common kernel for modules.
+	cmd, err = a.makeCmd(params, "bzImage", "modules", "prepare-objtool")
+	if err != nil {
+		return fmt.Errorf("failed to create command to make bzImage: %v", err)
+	}
+	cmd.Dir = commonKernelDir
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to make bzImage: %v", err)
+	}
+
+	// Install common modules.
+	moduleStagingDir := filepath.Join(commonKernelDir, "staging")
+	moduleInstallFlag := fmt.Sprintf("INSTALL_MOD_PATH=%v", moduleStagingDir)
+	cmd, err = a.makeCmd(params, "bzImage", moduleInstallFlag, "modules_install")
+	if err != nil {
+		return fmt.Errorf("failed to create command to install modules: %v", err)
+	}
+	cmd.Dir = commonKernelDir
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to install modules: %v", err)
+	}
+	return nil
+}
+
+func (a android) buildExtModules(params Params) error {
+	commonKernelDir := filepath.Join(params.KernelDir, "common")
+
+	// Location of external modules relative to common kernel dir.
+	mFlag := fmt.Sprintf("M=../%v", modulePath)
+	// Absolute location of the kernel source directory.
+	srcFlag := fmt.Sprintf("KERNEL_SRC=%v", commonKernelDir)
+
+	moduleStagingDir := filepath.Join(commonKernelDir, "staging")
+	moduleInstallFlag := fmt.Sprintf("INSTALL_MOD_PATH=%v", moduleStagingDir)
+
+	// Make external modules.
+	cmd, err := a.makeCmd(params, "-C", modulePath, mFlag, srcFlag, moduleInstallFlag)
+	if err != nil {
+		return fmt.Errorf("failed to create command to make external modules modules: %v", err)
+	}
+	cmd.Dir = params.KernelDir
+	cmd.Env = append([]string{}, moduleEnvVar)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to make external modules: %v", err)
+	}
+
+	// Install modules.
+	cmd, err = a.makeCmd(params, "-C", modulePath, mFlag, srcFlag, moduleInstallFlag, "modules_install")
+	if err != nil {
+		return fmt.Errorf("failed to create command to install modules: %v", err)
+	}
+	cmd.Dir = params.KernelDir
+	cmd.Env = append([]string{}, moduleEnvVar)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to install modules: %v", err)
+	}
+
+	return nil
+}
+
 func (a android) build(params Params) (ImageDetails, error) {
 	var details ImageDetails
+	var err error
```

Comment:
Nit: there's no need to declare a variable much ahead of its use.

Reply:
Makes sense; moved